src/OrbitGl/ScopeTreeTest.cpp

```diff
@@ -28,6 +29,40 @@ TestScope* CreateScope(uint64_t start, uint64_t end) {
   return scope_buffer.back().get();
 }
 
+uint64_t GetFakeTimeStamp() {
+  static uint64_t count = 0;
+  return ++count;
+}
+
+struct ScopeTimer {
+  ScopeTimer(std::vector<TestScope*>* scopes, size_t max_nodes) {
+    start = GetFakeTimeStamp();
+    max_num_nodes = max_nodes;
+    scope_buffer = scopes;
+  }
+  ~ScopeTimer() {
+    if (scope_buffer->size() < max_num_nodes) {
+      scope_buffer->push_back(CreateScope(start, GetFakeTimeStamp()));
+    }
+  }
+  uint64_t start;
+  size_t max_num_nodes;
+  std::vector<TestScope*>* scope_buffer;
+};
+
+void CreateNonOverlappingTestTree(size_t max_num_nodes, size_t max_depth,
+                                  std::vector<TestScope*>* scope_buffer, size_t depth = 0) {
+  if (depth > max_depth) return;
+  if (scope_buffer->size() >= max_num_nodes) return;
+
+  ScopeTimer timer(scope_buffer, max_num_nodes);
+  constexpr int kNumSiblingNodes = 4;
```

Comment:
Why not having this as a parameter also?

Reply:
Done. Thanks.